starboard/android/shared/system_egl.cc

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES2/gl2.h>
+
+#include "starboard/egl.h"
+#include "starboard/gles.h"
+
+#if !defined(EGL_VERSION_1_0) || !defined(EGL_VERSION_1_1) || \
+    !defined(EGL_VERSION_1_2) || !defined(EGL_VERSION_1_3) || \
+    !defined(EGL_VERSION_1_4)
+#error "EGL version must be >= 1.4"
+#endif
+
+namespace {
+
+bool first_make_current = false;
+
+EGLBoolean SbEglInitialize(EGLDisplay dpy, EGLint* major, EGLint* minor) {
+  first_make_current = true;
+  return eglInitialize(dpy, major, minor);
+}
+
+EGLBoolean SbEglTerminate(EGLDisplay dpy) {
+  first_make_current = false;
+  return eglTerminate(dpy);
+}
+
+EGLBoolean SbEglMakeCurrent(EGLDisplay dpy,
+                            EGLSurface draw,
+                            EGLSurface read,
+                            EGLContext ctx) {
+  EGLBoolean result = eglMakeCurrent(dpy, draw, read, ctx);
+  if (first_make_current && (dpy != EGL_NO_DISPLAY) &&
+      (draw != EGL_NO_SURFACE) && (eglGetError() == EGL_SUCCESS)) {
+    first_make_current = false;
+    // Start by showing a black surface immediately.
+    const SbGlesInterface* gles = SbGetGlesInterface();
+    gles->glClearColor(0, 0, 0, 1);
```

Comment:
I still don't understand why we need that extra draw. ClearNativeWindow() has been removed, so I think this is used by graphics system?  The old code only draws black while destroying player. Why does the new code need to draw black at app launch?

Reply:
The old code also calls `clearNativeWindow` from 
`Java_dev_cobalt_media_VideoSurfaceView_nativeOnVideoSurfaceChanged`. which was added for 
launches with deep links (b/199111478), with the comment 'Let the SurfaceView show a black screen immediately after the Surface is created'. This implements the same functionality, but earlier and more efficiently.

Reply:
I see, that's right.